engine/lite/request-response/runtime/src/main/scala/pl/touk/nussknacker/engine/requestresponse/openapi/RequestResponseOpenApiGenerator.scala

```diff
@@ -1,47 +1,94 @@
 package pl.touk.nussknacker.engine.requestresponse.openapi
 
+import akka.http.scaladsl.model.{HttpHeader, HttpRequest}
+import com.typesafe.scalalogging.LazyLogging
 import io.circe.Json
 import io.circe.syntax._
-import pl.touk.nussknacker.engine.requestresponse.RequestResponseInterpreter.RequestResponseScenarioInterpreter
 import pl.touk.nussknacker.engine.requestresponse.api.openapi.RequestResponseOpenApiSettings.OPEN_API_VERSION
 import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
 
-import scala.language.higherKinds
+import scala.collection.JavaConverters._
 
-object RequestResponseOpenApiGenerator {
+class RequestResponseOpenApiGenerator(oApiInfo: OApiInfo, serversDescription: Option[OApiServer], additionalInvocationPaths: List[String],
+                                      invocationPathFromDefinitionPath: List[String] => List[String]) extends LazyLogging {
 
-  private val jsonEncoder = BestEffortJsonEncoder(failOnUnkown = true, getClass.getClassLoader)
+  def generateOpenApiDefinition(request: HttpRequest, pathDefinitionGenerator: PathOpenApiDefinitionGenerator): Json = {
+    generateOpenApiForPathList(List(toInvocationPath(request) -> pathDefinitionGenerator) ++ additionalInvocationPaths.map(_ -> pathDefinitionGenerator))
+  }
 
-  def generateOpenApi[Effect[_]](pathWithInterpreter: List[(String, RequestResponseScenarioInterpreter[Effect])], oApiInfo: OApiInfo, serversDescription: Option[OApiServer]): String = {
-    val scenarioDefinitions: Json = generateScenarioDefinitions(pathWithInterpreter)
-    OApiDocumentation(OPEN_API_VERSION, oApiInfo, serversDescription.map(List(_)), scenarioDefinitions).asJson.spaces2
+  private def toInvocationPath(request: HttpRequest): String = {
+    val originalPathSegmentsOpt = request.headers.collectFirst {
+      // It can be configured using https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#proxy-add-original-uri-header
+      case HttpHeader("x-original-uri", originalPath) =>
```

Comment:
Is it standard by any means? e.g. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter#x-envoy-original-path
etc. Maybe it should be also configurable?